### DISCUSSION TITLE
Core team issue prioritization

### DISCUSSION THREAD

**Discussion by rndquu** [ubiquity/ubiquibot#398] [Ideas]
Right now bounty hunters and team members select issues to solve looking primarily on a bounty price. Our current bounty pricing structure basically mimics [value vs complexity](https://www.productplan.com/glossary/value-vs-complexity/) prioritization strategy which is totally fine. Time labels represent complexity while priority labels (as far as I understand) represent business value.

It is fine that outside collaborators (i.e. bounty hunters) take issues that they are able to solve. In the meantime the core team (i.e. organisation members) is aware of the whole product much better than outside bounty hunters. So it would be better if the core team (ideally along with outside collaborators) was primarily focused on the issues with high business value. If we continue to solve ["could have"](https://www.productplan.com/glossary/moscow-prioritization/) issues then we will simply run out of money.

Part of the team is on a payroll so we can set issue priorities for them. We could add a new `Roadmap` label which would basically mean a ["must have"](https://www.productplan.com/glossary/moscow-prioritization/) feature that would move the product to its final state (for which users are willing to pay money).

Examples of "could have" features (low business priority, good for outside bounty hunters):
- https://github.com/ubiquity/ubiquibot/issues/396
- https://github.com/ubiquity/ubiquity-dollar/issues/624

Examples of "must have" features (high business priority, should be the main focus of the core team):
- https://github.com/ubiquity/ubiquibot/issues/176
- https://github.com/ubiquity/ubiquity-dollar/issues/658

We could also build a roadmap based on the issues labeled with `Roadmap`.

So, simply put, I'm trying to find a way to focus the core team on the issues that moves us to:
1. ready to use bot app so we could charge partners for using it
2. production ready `UAD`

**Comment by 0x4007** [comment]
> So it would be better if the core team (ideally along with outside collaborators) was primarily focused on the issues with high business value.

Its a good proposal and I think that our leaderboards/experience points will be able to help quite a bit with this, where for example, new bounty hunters can only handle `Priority: 0` tasks when they start, and after collecting X amount of experience points (could be dollars collected) to be able to handle the next tier of priority. 

This will naturally promote bounty hunters to privileged positions of being able to handle higher priority items as they learn more about the codebase and rank up through our system.

Original post on the idea is here https://github.com/ubiquity/ubiquibot/discussions/71#discussioncomment-5643654

I know we had some additional conversation around this topic within the research team. I believe the conclusion was to use the total dollars collected to calculate their experience points (this means that for a new bounty hunter to rank up faster, they can complete the most complex issues, of which they are allowed to choose - `Priority: 0` issues.)

This models "core team" as a continuum vs a boolean which I think is more elegant.

The "very core" team can handle the highest priority items with the most amount of complexity, for example. The "kind of core team" can handle the middle ground tasks, while the outsiders, or new bounty hunters, can only handle the bottom tier tasks.

---

> We could add a new Roadmap label which would basically mean a ["must have"](https://www.productplan.com/glossary/moscow-prioritization/) feature that would move the product to its final state (for which users are willing to pay money).

There is a built in feature for milestones, is this suitable for what you're proposing? To be honest this idea of using a `Roadmap` label is a bit less clear to me.

**Reply by rndquu** [reply]
> There is a built in feature for milestones, is this suitable for what you're proposing?

Yes, it is. So we can simply create a new milestone and add "must have" features there so that the core team was aware of the features that should be focused.

**Reply by 0x4007** [reply]
Feel free to create that and we can iterate as needed.
